src/components/ComplexView.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {
+    ChangeEventHandler,
+    EventHandler,
+    FormEventHandler,
+    MouseEventHandler,
+    SyntheticEvent,
+    useCallback
+} from "react";
+import ReactDOM from "react-dom";
+import {shallow} from "enzyme";
+
+type Props = { children: (arg: any) => [object, object] };
+type State = {
+    value: boolean
+};
+
+class ComplexView extends React.Component<Props, State> {
+    changeview: any;
+    changestate: any
+    checkChildrenCallback: any;
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            value: false,
+        };
+
+        this.changestate = () => {
```

Comment:
make this either a method or just class field:
https://javascript.info/class#class-fields

Reply:
Maybe it would be nice to rename it to switchView or something like this?

Reply:
Done